src/engine/engineworkerscheduler.cpp

```diff
@@ -45,17 +42,19 @@ void EngineWorkerScheduler::runWorkers() {
 void EngineWorkerScheduler::run() {
     while (!m_bQuit) {
         Event::start("EngineWorkerScheduler");
-        EngineWorker* pWorker = NULL;
-        while (m_scheduleFIFO.read(&pWorker, 1) == 1) {
-            if (pWorker) {
-                pWorker->wake();
+        {
```

Comment:
You don't need to introduce separate scopes here, just use unlock() and relock().

Reply:
do you mean: 
```
    while (!m_bQuit) {
        Event::start("EngineWorkerScheduler");
        QMutexLocker lock(&m_mutex);
        for(const auto& pWorker: m_workers) {
            pWorker->wakeIfReady();
        }
        lock.unlock();
        Event::end("EngineWorkerScheduler");
        lock.relock();
        if (!m_bQuit) {
            // Wait for next runWorkers() call
            m_waitCondition.wait(&m_mutex); // unlock mutex and wait
        }
    }
```
I can do this, but it looks less nice for me.

Reply:
I'm fine with both.

Reply:
Than I like to s tick with the current version. 
Anything else to do?